File: src/test/java/jt/ExampleTest.java
package jt;

import me.saro.selenium.ChromeDriverManager;
import me.saro.selenium.model.DownloadStrategy;
import me.saro.selenium.model.Platform;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

//@Disabled
public class ExampleTest {

    @Test
    public void test() {
        ChromeDriverManager manager = ChromeDriverManager.builder(new File("./tmp"))
                .enableRecommendChromeOptions(true)
                .build();

        List<String> list = manager.openBackground("https://gs.saro.me", dp -> {
            List<String> items = new ArrayList<>();
            dp.finds(".post-list .node").forEach(
                    e -> items.add(dp.find(e, "a").getText().trim() + " " + dp.find(e, "a").getAttribute("href"))
            );
            return items;
        });

        list.forEach(System.out::println);
    }

    @Test
    public void download() {
        ChromeDriverManager.download(new File("./tmp"), Platform.getPlatform(), DownloadStrategy.DOWNLOAD_IF_NO_VERSION);
    }
}

File: src/test/kotlin/kt/ExampleTest.kt
package kt

import me.saro.selenium.ChromeDriverManager
import me.saro.selenium.model.DownloadStrategy.DOWNLOAD_IF_NO_VERSION
import me.saro.selenium.model.Platform.Companion.getPlatform
import org.junit.jupiter.api.Test
import java.io.File

// @Disabled
class ExampleTest {
    @Test
    fun test() {
        val manager = ChromeDriverManager.builder(File("./tmp"))
            .downloadStrategy(DOWNLOAD_IF_NO_VERSION) // default value
            .enableRecommendChromeOptions(true)
            .build()

        val list = manager.openBackground("https://gs.saro.me") {
            finds(".post-list .node")
                .map { it.find("a").text.trim() + " : " + it.find("a").getAttribute("href") }
        }

        list.forEach(::println)
    }

    @Test
    fun download() {
        ChromeDriverManager.download(File("./tmp"), getPlatform(), DOWNLOAD_IF_NO_VERSION)
    }
}